lib/internal/http2/compat.js

```diff
@@ -582,9 +581,7 @@ class Http2ServerResponse extends Stream {
 
   addTrailers(headers) {
     const keys = ObjectKeys(headers);
-    let key = '';
-    for (let i = 0; i < keys.length; i++) {
-      key = keys[i];
+    for (const key of keys) {
```

Comment:
`for(;;)` loops are preferred for performance reasons. Also, iterating over an array relies on the globally mutable `Array.prototype[Symbol.iterator]`